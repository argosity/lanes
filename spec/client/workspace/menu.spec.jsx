import React from 'react';
import { each, range } from 'lodash';

import Group      from 'lanes/screens/group';
import Menu       from 'lanes/workspace/menu';
import MenuGroup  from 'lanes/workspace/menu-group';
import MenuOption from 'lanes/workspace/menu-option';

import { Snapshot } from '../test-utils';
import { getTestScreen } from '../test-models';

describe('Workspace Menu', () => {
    beforeEach(() => {
        each(range(0, 5), i =>
            Group.register({ id: `group-${i}`, title: `Group # ${i}`, icon: `grp-${i}` }),
        );
    });

    it('renders with a logout button', () => {
        const menu = shallow(<Menu />);
        expect(menu).toHaveRendered('Logout');
    });

    it('renders groups', () => {
        const menu = shallow(<Menu />);
        expect(menu.find('Group').length).toEqual(5);
        expect(Snapshot(<Menu />)).toMatchSnapshot();
    });

    it('has groups with screens', () => {
        const group = shallow(<MenuGroup group={Group.all[0]} />);
        expect(group).not.toHaveRendered('MenuOption');

        getTestScreen({ id: 'new', group_id: Group.all[0].id });
        expect(group).toHaveRendered('MenuOption');
        expect(Snapshot(<MenuGroup group={Group.all[0]} />)).toMatchSnapshot();
    });

    it('has renders a menu option', () => {
        const screen = getTestScreen();
        const option = shallow(<MenuOption screen={screen} />);
        expect(option).toHaveRendered(`Anchor[path="/${screen.id}/"]`);
        expect(option).toHaveRendered(`Icon[type="${screen.icon}"]`);
        expect(Snapshot(<MenuOption screen={screen} />)).toMatchSnapshot();
    });
});
