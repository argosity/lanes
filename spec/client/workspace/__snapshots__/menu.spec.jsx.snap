// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace Menu has groups with screens 1`] = `
<nav
  className="grommetux-box grommetux-box--direction-column grommetux-box--justify-between grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none grommetux-menu grommetux-menu--column grommetux-menu--primary grommetux-menu--inline"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--flex-off"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <header
      className="grommetux-box grommetux-box--direction-row grommetux-box--align-end grommetux-box--pad-horizontal-medium grommetux-header"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}
    >
      <h4
        className="grommetux-heading grommetux-heading--strong"
      >
        Settings
        <span
          aria-hidden={true}
          className="fa fa-wrench icon"
        />
      </h4>
    </header>
  </div>
  <a
    aria-label={undefined}
    className="grommetux-anchor"
    href={undefined}
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-heartbeat icon"
    />
    Preferences
  </a>
  <a
    aria-label={undefined}
    className="grommetux-anchor"
    href={undefined}
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-group icon"
    />
    User Management
  </a>
  <a
    aria-label={undefined}
    className="grommetux-anchor"
    href={undefined}
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-cogs icon"
    />
    System Settings
  </a>
  <a
    aria-label={undefined}
    className="grommetux-anchor"
    href={undefined}
    onClick={[Function]}
  >
    <span
      aria-hidden={true}
      className="fa fa-red icon"
    />
    title
  </a>
</nav>
`;

exports[`Workspace Menu has renders menu options 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor"
  href={undefined}
  onClick={[Function]}
>
  <span
    aria-hidden={true}
    className="fa fa-red icon"
  />
  title
</a>
`;
